src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -307,6 +307,20 @@ private static bool IsFloatPointExpressionOfUnknownPrecision(BoundExpression rew
 
                     break;
 
+                case ConversionKind.MethodGroup:
+
+                    if (_inExpressionLambda || explicitCastInCode || (object)symbolOpt == null || !symbolOpt.IsStatic || symbolOpt.MethodKind != MethodKind.Ordinary)
+                    {
+                        break;
+                    }
+                    if (symbolOpt.IsExtensionMethod && ((BoundMethodGroup)rewrittenOperand).ReceiverOpt.Kind != BoundKind.TypeExpression)
+                    {
+                        break;
+                    }
+
+                    return RewriteMethodGroupConversion(oldNode, syntax, rewrittenOperand, symbolOpt, isExtensionMethod, rewrittenType);
```

Comment:
Actually, I don't know! I wrote `symbolOpt.IsExtensionMethod` originally but failed Emit, from the debugger, I observe that `isExtensionMethod` looks like a mysterious `false` value came from the rewriter visit that seems to have nothing to do with `symbolOpt.IsExtensionMethod`. So I chose to preserve whatever rewritten value came out.